src/Query/Filter/TextField.ts

```diff
@@ -66,7 +72,59 @@ export abstract class TextField extends Field {
      */
     public abstract value(task: Task): string;
 
-    public static maybeNegate(match: boolean, filterMethod: String) {
-        return filterMethod.match(/not/) ? !match : match;
+    protected getFilter(matcher: IStringMatcher, negate: boolean): Filter {
+        return (task: Task) => {
+            const match = matcher!.matches(this.value(task));
+            return negate ? !match : match;
+        };
+    }
+}
+
+/**
+ * MultiTextField is an abstract base class to help implement
+ * all the filter instructions that act on multiple string values
+ * such as the tags.
+ */
+export abstract class MultiTextField extends TextField {
```

Comment:
The convention in this project is one class per file, generally speaking. 

So please could the new class be separated out to a new file? Thank you.

Reply:
That will also help with the review, as I can diff the pairs of classes.